packages/mask/src/plugins/Trader/messages.ts

```diff
@@ -1,12 +1,16 @@
 import { createPluginMessage, createPluginRPC, PluginMessageEmitter } from '@masknet/plugin-infra'
 import type { GasConfig } from '@masknet/web3-shared-evm'
+import type { SourceType } from '@masknet/web3-shared-base'
 import { PLUGIN_ID } from './constants/index.js'
 import type { TagType } from './types/index.js'
 
-interface CashTagEvent {
+interface CashAnchorEvent {
     name: string
     type: TagType
-    element: HTMLAnchorElement | null
+    element: HTMLElement | null
+    dataProviders?: SourceType[]
+    address?: string
+    isNFTProjectPopper?: boolean
```

Comment:
How can a cash anchor event collect that information?

Reply:
renamed cashAnchorObserved => trendingAnchorObserved

Reply:
Not the naming problem. It's about we cannot grab that information from the cash link.